DNN Platform/Library/Services/Mail/OAuth/ISmtpOAuthProvider.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.Services.Mail.OAuth
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    using MailKit.Net.Smtp;
```

Comment:
This implementation/reference makes the Oauth implementation no longer generic for both `MailKitMailProvider` and `CoreMailProvider` as this introduces a dependency on MailKit, is this expected?

Reply:
I think this is expected, as the oauth provider only compatible with Mailkit provider.

Reply:
@mitchelsellers I've resolved this dependency issue by introducing a new interface to wrap the MailKit SMTP client (hopefully it's sufficiently generic enough to support some other implementation).

Reply:
@bdukes You are awesome!